Feature/forms (#254)
infinum/eightshift-docs: website/forms/javascript/events.mdx

---
id: events
title: Events
---

You can listen to these events by using the `on` method on the `Event` facade.

### esFormsBeforeFormSubmit

Triggered before form is submitted to the API-Rest endpoint.

:::tip
This event is set on the form element.
:::

### esFormsAfterFormSubmit

Triggered after form is submitted to the API-Rest endpoint but before the response is processed.

:::tip
This event is set on the form element.
:::

### esFormsAfterFormSubmitSuccess

Triggered after form is submitted to the API-Rest endpoint and the response is processed with success.

:::tip
This event is set on the form element.
:::

### esFormsAfterFormSubmitError

Triggered after form is submitted to the API-Rest endpoint and the response is processed with general error.

:::tip
This event is set on the form element.
:::

### esFormsAfterFormSubmitErrorValidation

Triggered after form is submitted to the API-Rest endpoint and the response is processed with validation error.

:::tip
This event is set on the form element.
:::

### esFormsAfterFormSubmitEnd

> Triggered after form is submitted to the API-Rest endpoint and the response is processed till the end and form is ready for another submit.

Reviewer: ```suggestion
Triggered after form is submitted to the API-Rest endpoint and the response is fully processed and the form is ready for another submit.
```

:::tip
This event is set on the form element.
:::

### esFormsAfterGtmDataPush

Triggered after GTM dataLAyer is pushed. This happens after form is submitted to the API-Rest endpoint and the response is processed with success.

:::tip
This event is set on the form element.
:::

### esFormsAfterFormSubmitReset

> Triggered after form is submitted to the API-Rest endpoint and the response is processed with success and form is reset to its original state.

Reviewer: ```suggestion
Triggered after form is submitted to the API-Rest endpoint and the response is processed with success after which the form is reset to its original state.
```

:::tip
This event is set on the form element.
:::

### esFormsAfterFormSubmitSuccessBeforeRedirect

Triggered after form is submitted to the API-Rest endpoint and the response is processed with success right before redirect action triggers.

:::tip
This event is set on the form element.
:::

### esFormsJsFormLoaded

> Triggered when all JavaScript on form is loaded and form is ready to be used.

Reviewer: ```suggestion
Triggered when all JavaScript on the form is loaded and form is ready to be used.
```

:::tip
This event is set on the window element.
:::

### esFormsManualInitLoaded

> Triggered when manual initialization option is used from the forms global settings and the form is ready to be maunual initialized.

Reviewer: ```suggestion
Triggered when manual initialization option is used from the forms global settings and the form is ready to be maunually initialized.
```

:::tip
This event is set on the window element.
:::

### esFormsAfterCaptchaInit

> Triggered after captcha is loaded every page set by the forms global settings page.

Reviewer: Not quite clear what is happening here, could you please re-phrase this?
